pkg/notify/discord.go

```diff
@@ -0,0 +1,187 @@
+package notify
+
+import (
+	"fmt"
+
+	"github.com/bwmarrin/discordgo"
+	"github.com/infracloudio/botkube/pkg/config"
+	"github.com/infracloudio/botkube/pkg/events"
+	log "github.com/infracloudio/botkube/pkg/logging"
+)
+
+var embedColor = map[events.Level]int{
+	events.Info:     8311585,  // green
+	events.Warn:     16312092, // yellow
+	events.Debug:    8311585,  // green
+	events.Error:    13632027, // red
+	events.Critical: 13632027, // red
+}
+
+// Discord contains URL and ClusterName
+type Discord struct {
+	Token     string
+	ChannelID string
+	NotifType config.NotifType
+}
+
+// NewDiscord returns new Discord object
+func NewDiscord(c *config.Config) Notifier {
+	return &Discord{
+		Token:     c.Communications.Discord.Token,
+		ChannelID: c.Communications.Discord.Channel,
+		NotifType: c.Communications.Discord.NotifType,
+	}
+}
+
+// SendEvent sends event notification to Discord url
+func (d *Discord) SendEvent(event events.Event) (err error) {
+	log.Logger.Debug(fmt.Sprintf(">> Sending to discord: %+v", event))
+
+	api, err := discordgo.New("Bot " + d.Token)
+	if err != nil {
+		log.Logger.Error("error creating Discord session,", err)
+		return err
+	}
+	messageSend := formatDiscordMessage(event, d.NotifType)
+
+	// log.Logger.Debugf("message Embed: %+v", attachment)
+
+	if _, err := api.ChannelMessageSendComplex(d.ChannelID, &messageSend); err != nil {
+		log.Logger.Errorf("Error in sending message: %+v", err)
+		return err
+	}
+	log.Logger.Debugf("Event successfully sent to channel %s", d.ChannelID)
+	return nil
+}
+
+// SendMessage sends message to Discord url
+func (d *Discord) SendMessage(msg string) error {
+	log.Logger.Debug(fmt.Sprintf(">> Sending to discord: %+v", msg))
+	api, err := discordgo.New("Bot " + d.Token)
+	if err != nil {
+		log.Logger.Error("error creating Discord session,", err)
+		return err
+	}
+
+	if _, err := api.ChannelMessageSend(d.ChannelID, msg); err != nil {
+		log.Logger.Error("Error in sending message:", err)
```

Comment:
Missing return error from here ?

Reply:
`Send()` in slack,discord and `sendMessage()` in mattermost are implemented in the similar way. do we need to restructure this?

Reply:
Ideally, we should return an error here and log it at caller. We should also include Send method in Bot interface. I've created an issue to track this https://github.com/infracloudio/botkube/issues/270

Reply:
can these changes go with a separate PR?. Can this be considered resolved.?

Reply:
Let's address this in separate PR

Reply:
thanks, @sanketsudake  and @PrasadG193 
marking this as resolved